doc/source/skips/4-transition-to-v2.rst

```diff
@@ -0,0 +1,309 @@
+argument, arriving, after a further two releases, at the same place:
+scikit-image no longer rescales data automatically, there are no
+unnecessary keyword arguments lingering all over the API.
+
+Of course, this kind of operation would have to be done simultaneously over all
+of the above proposed changes.
+
+Ultimately, the core team felt that this approach generates more work for both
+the scikit-image developers and the developers of downstream libraries, for
+dubious benefit: ultimately, later versions of scikit-image will still be
+incompatible with prior versions, although over a longer time scale.
+
+A single package containing both versions
+.........................................
+
+Since the import name is changing, it would be possible to make a single
+package with both the ``skimage`` and ``skimage2`` namespaces shipping
+together, at least for some time. This option is attractive but it implies
+longer-term maintenance of the 1.0 namespace, for which we might lack
+maintainer time, or a long deprecation cycle for the 1.0 namespace, which would
+ultimately result in a lot of unhappy users getting deprecation messages from
+their scikit-image use.
+
+Not making the proposed API changes
+...................................
+
+Another possibility is to reject backwards incompatible API changes outright,
+except in extreme cases. The core team feels that this is essentially
+equivalent to pinning the library at 0.19.
+
+"scikit-image2" as the new package name
+.......................................
+
+The authors acknowledge that the new names should be chosen with care to keep
+the disruption to scikit-image's user base and community as small as possible.
+However, to protect users without upper version constraints from accidentally
+upgrading to the new API, the package name ``scikit-image`` must be changed.
+Changing the import name ``skimage`` is similarly advantageous because it allows
+using both APIs in the same environment.
+
+This document suggests just ``skimage2`` as the single new name for
+scikit-image's API version 2.0, both for the import name and the name on PyPI,
+conda-forge and elsewhere. The following arguments were given in favor of this:
+
+- Only one new name is introduced with the project thereby keeping the number of
+  associated names as low as possible.
+- With this change, the import and package name match.
+- Users might be confused whether they should install ``scikit-image2`` or
+  ``scikit-image-2``. It was felt that ``skimage2`` avoids this confusion.
+- Users who know what ``skimage`` is and see ``skimage2`` in an install
+  instruction somewhere, will likely be able to infer that it is a newer version
+  of the package.
+- It is unlikely that users will be aware of the new API 2.0 but not of the new
+  package name. A proposed release of scikit-image 1.1 might point users to
+  ``skimage2`` during the installation and update process and thereby clearly
+  communicate the successors name.
```

Comment:
I really like @grlee77's suggestion to release a scikit-image 1.1 version that advises users to either pin scikit-image to 1.0 to avoid a warning or upgrade to v2 by installing "skimage2". So I optimistically included the talking point here. 

@jni this point should probably be explained and described earlier in the implementation section. I'm happy to do that tomorrow unless someone beats me to it or that this point proves controversal.